onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -5139,10 +5139,25 @@ def aten_square(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_squeeze(self: TensorType) -> TensorType:
+@torch_op("aten::squeeze", trace_only=True)
+def aten_squeeze(self: TTensor, dim: Optional[int] = None) -> TTensor:
     """squeeze(Tensor(a) self) -> Tensor(a)"""
 
-    raise NotImplementedError()
+    if op.OptionalHasElement(dim):
+        rank = op.Size(op.Shape(self))
+        if rank == 0:
+            self = op.Reshape(self, op.Constant(value_ints=[-1]))
+        # check if specified dimension equal to 1
+        shape = op.Shape(self, start=dim, end=dim + 1)  # type: ignore[operator]
+        if shape != 1:
```

Comment:
Is this an error-condition or acceptable in aten's squeeze op?

Reply:
aten_squeeze definition: when value_of_specified_dim != 1, don't do the squeeze, just return self.